Palindrome.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+
+namespace Palindrome
+{
+    public sealed class Palindrome
+    {
+        /// <summary>
+        /// Method checks if input string is a palindrome.
+        /// Altered method to accept string input instead of char Array.
+        /// Functionally they are equivalent.
+        /// Modified to execute without duplicating strings using 
+        /// minimum space complexity.
+        /// </summary>
+        /// <param name="arr">A character array.</param>
+        /// <returns>Boolean result</returns>
+        public bool IsPalindrome(string strInput)
```

Comment:
Typically, you don't want to include the data type in the string (i.e. `str`). Input is fine.